### PR TITLE
TestE2E GetCollectedTimeByName

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Return the timestamp when the payloads has been collected aim to check if it was received as expected.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/fakeintake/aggregator/common.go [hunk 1]
```diff
@@ -23,8 +24,9 @@ type PayloadItem interface {
 type parseFunc[P PayloadItem] func(payload api.Payload) (items []P, err error)
 
 type Aggregator[P PayloadItem] struct {
-	payloadsByName map[string][]P
-	parse          parseFunc[P]
+	payloadsByName    map[string][]P
+	collectedAtByName map[string][]time.Time
```
Comment (pducolin) [inline test/fakeintake/aggregator/common.go:28]: **❓ question**
What about adding `GetCollectedTime` to `PayloadItem` interface ? This implementation makes it hard to know at which moment one specific payload was received

Comment (nplanel) [reply test/fakeintake/aggregator/common.go]: Thanks, changing the interface was the right way

**Review by pducolin — APPROVED** [review]
Nice ! Thank you !

**Comment by pducolin** [comment]
Interestingly, aggregator tests are failing only on windows. I wonder if we skip them on linux.

**Comment by nplanel** [comment]
> Interestingly, aggregator tests are failing only on windows. I wonder if we skip them on linux.

and more interessting windows linter should not failed as that was my last fix to add comments ...
Seems linux/windows linter have different config

**Comment by nplanel** [comment]
> > Interestingly, aggregator tests are failing only on windows. I wonder if we skip them on linux.
> 
> and more interessting windows linter should not failed as that was my last fix to add comments ... Seems linux/windows linter have different config

github didn't refresh the jobs https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/20058950

**Pull request merged**